Code/Source/RobotImporter/RobotImporterWidget.cpp

```diff
@@ -27,45 +28,82 @@ namespace ROS2
         mainLayout->addWidget(captionLabel);
         QPushButton* selectFileButton = new QPushButton(QObject::tr("Load"), this);
         mainLayout->addWidget(selectFileButton);
-        mainLayout->addWidget(&m_robotFileNameLabel);
-        mainLayout->addWidget(&m_robotNameLabel);
-
-        m_importFileDialog.setFileMode(QFileDialog::ExistingFiles);
-        m_importFileDialog.setNameFilter(QObject::tr("Unified Robot Description Format (*.urdf)"));
-        m_importFileDialog.setViewMode(QFileDialog::Detail);
+        mainLayout->addWidget(&m_logLabel);
+        mainLayout->addStretch();
 
         QObject::connect(
             selectFileButton,
             &QPushButton::clicked,
             this,
             [this]()
             {
-                int result = m_importFileDialog.exec();
-                if (result != QDialog::DialogCode::Accepted)
-                {
-                    return;
-                }
-
-                auto fileNameToImport = m_importFileDialog.selectedFiles().first();
-                m_robotFileNameLabel.setText(fileNameToImport);
-                auto fileNameToImportAZ(fileNameToImport.toUtf8().constData());
-                m_urdfModel = UrdfParser::ParseFromFile(fileNameToImportAZ);
-                OnModelLoaded();
+                m_robotImporter.Import();
             });
-
-        mainLayout->addStretch();
         setLayout(mainLayout);
     }
 
-    void RobotImporterWidget::OnModelLoaded()
+    void RobotImporterWidget::ReportError(AZStd::string errorMessage)
     {
-        m_robotNameLabel.setText(m_urdfModel->getName().c_str());
+        QMessageBox::critical(this, QObject::tr("Error"), errorMessage.c_str());
+        m_logLabel.setText(errorMessage.c_str());
+        AZ_Error("RobotImporterWidget", false, errorMessage.c_str());
+    }
+    void RobotImporterWidget::ReportWarning(AZStd::string warningMessage)
+    {
```

Comment:
No text is set for log label here. Is this intentional?
It would be best to extract this behavior to a dedicated utility class. Is there anything similar in O3DE already?

Reply:
Warning reporting was created to make a more "complete" reporting API (error, warn, info). Since I did not use it anywhere I got rid of it entirely. When it will be needed it can be brought back. Some more sophisticated warning reporting should be placed in GUI in the future.